h2o-r/h2o-package/R/models.R

```diff
@@ -3138,21 +3218,103 @@ h2o.deepfeatures <- function(object, data, layer) {
 setClass(
   "H2OTree",
   representation(
+    root_node = "H2ONode",
     left_children = "integer",
     right_children = "integer",
+    node_ids = "integer",
     descriptions = "character",
     model_id = "character",
     tree_number = "integer",
     tree_class = "character",
-    root_node_id = "integer",
     thresholds = "numeric",
     features = "character",
     levels = "list",
-    nas = "character"
+    nas = "character",
+    predictions = "numeric"
   )
-  
 )
 
+#' @rdname H2OTree-class
+#' @param object an \code{H2OTree} object.
+#' @export
+setMethod('show', 'H2OTree', 
+          function(object){
+            print.H2OTree(object)
+          })
+
+print.H2OTree <- function(tree){
+  cat(paste0("Tree related to model '", tree@model_id,"'. Tree number is"), paste0(tree@tree_number,", tree class is '",tree@tree_class, "'\n"))
+  cat("The tree has", length(tree), "nodes")
+}
+
+#'
+#' Overrides the behavior of length() function on H2OTree class. Returns number of nodes in an \code{H2OTree}
+#' @param x An \code{H2OTree} to count nodes for.
+#'
+setMethod("length", signature(x = "H2OTree"), function(x) {
+  length(x@left_children)
+})
+
+
+
+.h2o.assemble_tree <- function(tree){
+  
+  left_child = .h2o.walk_tree(tree@left_children[1], tree)
+  right_child = .h2o.walk_tree(tree@right_children[1], tree)
+  
+  rootNode <- NULL
+  
+  if(is.null(left_child) && is.null(right_child)){
```

Comment:
is there a reason the code for the inner nodes cannot be reused for the root node as well?